include/class.thread.php

```diff
@@ -993,14 +993,19 @@ function lookupByEmailHeaders(&$mailinfo, &$seen=false) {
                 //Lookup by ticket number
                 && ($ticket = Ticket::lookupByNumber($match[1]))
                 //Lookup the user using the email address
-                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))) {
+                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))
+                && ($staff = Staff::getIdByEmail($mailinfo['email']))) {
```

Comment:
Perhaps I should have slept at a Holiday Inn Express last night; but the `User::lookup` should definitely fail if no end user currently exists with that email address. Perhaps your staff members already all have corresponding user accounts?